### ISSUE TITLE
No (Obvious) way to create named terminals

### ISSUE THREAD

**Issue by cancan101** [jupyter/notebook#2602]
Jupyter notebook supports naming the terminals:
![image](https://user-images.githubusercontent.com/51059/27503601-80b868ea-584b-11e7-90e9-960c968526b5.png)

however there seems to be no UI to either set the terminal name or to rename the terminal once it has been created. The only way I have found to create a named terminal is to browse to: `http://<host:port>/terminals/<terminal name>`. By default the create new terminal menu creates numbered terminals:
![image](https://user-images.githubusercontent.com/51059/27503622-b6ad2b3e-584b-11e7-8a7a-a1d868625ca4.png)

Further, on the terminal page itself, there is no way to see the terminal name other than to look at the location bar.

**Comment by takluyver** [comment]
Yep, it's functionality from terminado, the library that manages serving terminals, that we don't really try to expose, but it's easier to leave it working than to disable it.

**Comment by Gilbert-Changky** [comment]
> Jupyter notebook supports naming the terminals:
> ![image](https://user-images.githubusercontent.com/51059/27503601-80b868ea-584b-11e7-90e9-960c968526b5.png)
> 
> however there seems to be no UI to either set the terminal name or to rename the terminal once it has been created. The only way I have found to create a named terminal is to browse to: `http://<host:port>/terminals/<terminal name>`. By default the create new terminal menu creates numbered terminals:
> ![image](https://user-images.githubusercontent.com/51059/27503622-b6ad2b3e-584b-11e7-8a7a-a1d868625ca4.png)
> 
> Further, on the terminal page itself, there is no way to see the terminal name other than to look at the location bar.

How did you name your terminals? Mine just open without any chance to name it.
